sql/log_event_client.cc

```diff
@@ -1761,9 +1761,9 @@ bool Log_event::print_base64(IO_CACHE* file,
           TODO: fix MDEV-10362 to remove this workaround.
         */
         if (print_event_info->base64_output_mode !=
-            BASE64_OUTPUT_DECODE_ROWS)
+            BASE64_OUTPUT_DECODE_ROWS && !is_more)
```

Comment:
Actually this method to continue printing the delimiter out to the output file 
assumes MDEV-10362 is not fixed yet, but the hunk's agenda is exactly it :-) (I duplicate it btw today on Jira).
So now with the new footer cache introduced, we just need two steps more:
1.this whole workaround block got to be dismantled in favor of
2.printing the delimiter unconditionally in 
copy_cache_to_string_wrapped() and copy_cache_to_string_wrapped().

Like I suggested on Jira MDEV-22338, let me carry out the above plan in MDEV-16372 (that is duplicated by MDEV-10362). [At this point I only have 10.1 patch that does not have this wa/ block].

Reply:
> Actually this method to continue printing the delimiter out to the output file
> assumes MDEV-10362 is not fixed yet, but the hunk's agenda is exactly it :-) (I duplicate it btw today on Jira).
> So now with the new footer cache introduced, we just need two steps more:
> 1.this whole workaround block got to be dismantled in favor of
> 2.printing the delimiter unconditionally in
> copy_cache_to_string_wrapped() and copy_cache_to_string_wrapped().
> 
> Like I suggested on Jira MDEV-22338, let me carry out the above plan in MDEV-16372 (that is duplicated by MDEV-10362). [At this point I only have 10.1 patch that does not have this wa/ block].

Thanks andrel, I added `footer_cache` according to MySQL's handling method for the --verbose option. The commented SQL saved by `footer_cache` will be printed after BASE64_OUTPUT_DECODE_ROWS, which solves this problem conveniently. 
But if you will continue to fix this problem later, I can remove the code related to bug01 in this patch.

Reply:
Yes, please leave out `footer_cache`. I am finishing off MDEV-16372 work with 10.2+ patch today.